openapi.yml

```diff
@@ -32,71 +55,85 @@ paths:
                 type: array
                 items:
                   $ref: "#/components/schemas/OrderFull"
-  /api/v1/nonce:
+  /api/v1/gas-tip/{sellToken}:
     get:
-      summary: The nonce to pick for a new order.
-      description: The best nonce to pick depends on how much time there is left in the current settlement according to the operator. This time is token-pair specific which is why the sell and buy tokens must be supplied.
-      parameters:
-        - name: sellToken
-          in: query
-          schema:
-            $ref: "#/components/schemas/Address"
-        - name: buyToken
-          in: query
-          schema:
-            $ref: "#/components/schemas/Address"
+      summary: The tip that is used to pay solvers for their service in the sellToken of an order.
+      description: The tip is calculated by the order service and depends on the current gas price and the expected gas costs to settle the trade
       responses:
         200:
-          description: the nonce
+          description: the tip
           content:
             application/json:
               schema:
-                $ref: "#/components/schemas/Nonce"
+                $ref: "#/components/schemas/Tip"
+        400:
+          description: sellToken not existent
 components:
   schemas:
     Address:
       description: Ethereum 40 byte address encoded as a hex without `0x` prefix.
       type: string
       example: "6810e776880c02933d47db1b9fc05908e5386b96"
     TokenAmount:
-      description: Amount of a token. uint112 encoded in decimal.
+      description: Amount of a token. uint256 encoded in decimal.
       type: string
       example: "1234567890"
-    Nonce:
-      description: Nonce of the batch in which the order is valid. uint32.
-      type: integer
-    OrderNew:
+    Tip:
+      description: The tip that is used to pay solvers. uint256.
+      type: object
+      properties:
+        expirationDate:
+          description: |
+            Expiration date of the offered tip. Order service might not accept 
+            the tip after this expiration date. Encoded as unix timestamp.
+          type: integer
+        tipAmount:
+          description: Amount of tip in specified sellToken
+          type: integer
```

Comment:
This should be a string

Reply:
I agree with your last comments. We should use ISO dates. It's a more natural format for dates, and easier to read by anyone

This case is different than for the order. In the order the expirationDates is an epoch in the contract, therefore is what the user signs. So the creation of the orders endpoints has the original data (so the epoch) and the signature of it. So it was justified

Reply:
I agree that here we have a different situation. Though I see value in keeping it always to unix time, as we have chosen it once.

But I really don't have a strong opinion.

Reply:
Yes is not a critical decision, my point is that, in doubt I always prefer readability. Specially if this is meant to be use for integrations with 3rd parties.

Some pros/cons to make up our your minds.
https://stackoverflow.com/questions/47426786/epoch-or-iso8601-date-format